packages/core-ethereum/src/indexer/index.ts

```diff
@@ -852,13 +870,17 @@ class Indexer extends (EventEmitter as new () => IndexerEventEmitter) {
     }
 
     let prevState: ChannelEntry
-    try {
-      prevState = await this.db.getChannel(channel.get_id())
-    } catch (e) {
-      // Channel is new
+    let channel_entry = await this.db.get_channel(Ethereum_Hash.deserialize(channel.get_id().serialize()))
+    if (channel_entry !== undefined) {
+      prevState = ChannelEntry.deserialize(channel_entry.serialize())
     }
 
-    await this.db.updateChannelAndSnapshot(channel.get_id(), channel, lastSnapshot)
+    assert(lastSnapshot !== undefined)
```

Comment:
This will crash the node, is that the desired behaviour? Can it recover upon restart from it?

Reply:
See above.